lms/djangoapps/instructor_task/models.py

```diff
@@ -101,6 +102,16 @@ def create(cls, course_id, task_type, task_key, task_input, requester):
         task_id = str(uuid4())
         json_task_input = json.dumps(task_input)
 
+        # check length of task_input, and return an exception if it's too long
+        if len(json_task_input) > TASK_INPUT_LENGTH:
+            error_msg = u'Task input longer than "{length}": "{input}" for "{task}" of "{course}"'.format(
+                length=TASK_INPUT_LENGTH,
+                input=json_task_input,
+                task=task_type,
+                course=course_id
+            )
+            raise ValueError(error_msg)
```

Comment:
What happens after this error is raised? Does it get logged somewhere, or bubble up to the user somehow?

Reply:
@ormsbee thanks for having a look. Your question made me realize that we should either show the error message to the user or log it (maybe, do both?). I have modified the code so that error is shown to the user as 
<img width="681" alt="Screen Shot 2019-08-06 at 1 24 42 PM" src="https://user-images.githubusercontent.com/22347092/62524055-64d2b080-b84e-11e9-9e06-1c6fac3993ea.png">
Please let me know if that makes sense.
Thanks.

Reply:
I definitely think it needs to be logged. It's probably not a useful error message to bubble up to the instructor, since it's unlikely that they'd draw the line between "Generate Problem Grade Report" and input length. Though the first thing they'd do is send this to the support team, so I guess there is that.

So basically: Please log it somewhere. I'm ambivalent about whether the message should bubble up to the user or be replaced with something more generic -- either seems okay to me. Thank you.

Reply:
@ormsbee I have just made the error message generic. It now looks as 
<img width="782" alt="Screen Shot 2019-08-07 at 1 34 59 PM" src="https://user-images.githubusercontent.com/22347092/62607965-68326e80-b918-11e9-94bd-939ade5e6b7d.png">
Also, logging the error now.
I have kept user-facing error message so that task creation is not failing silently and the user is notified about the failure.
Thanks.